### ISSUE TITLE
Container crashes - Permission denied

### ISSUE THREAD

**Issue by adrianchifor** [verdaccio/verdaccio#238]
```
fs.js:641
  return binding.open(pathModule._makeLong(path), stringToFlags(flags), mode);
                 ^

Error: EACCES: permission denied, open '/verdaccio/storage/.sinopia-db.json'
    at Error (native)
    at Object.fs.openSync (fs.js:641:18)
    at Object.fs.writeFileSync (fs.js:1347:33)
    at LocalData.sync (/usr/local/app/lib/storage/local/local-data.js:67:8)
    at new Storage (/usr/local/app/lib/storage.js:31:33)
    at module.exports (/usr/local/app/lib/index.js:17:19)
    at afterConfigLoad (/usr/local/app/lib/cli.js:115:15)
    at Object.<anonymous> (/usr/local/app/lib/cli.js:66:1)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
```

Started happening since we upgraded to 2.2.2. Looks like `.sinopia-db.json` is created by root and is then accessed by verdaccio user (introduced in 2.2.2).

**Comment by adrianchifor** [comment]
If it helps, we run verdaccio inside Kubernetes and the `/verdaccio/storage` path is mounted on an AWS EBS volume.

**Comment by juanpicado** [comment]
Do you think this should be enough to fix it?
```
RUN  chown -R verdaccio:verdaccio /verdaccio/storage && \
     chown -R verdaccio:verdaccio /verdaccio/conf/
````

**Comment by adrianchifor** [comment]
Looks like this is already in the Dockerfile
```
RUN addgroup -S verdaccio && adduser -S -g verdaccio verdaccio && \
    chown -R verdaccio:verdaccio "$APPDIR" && \
    chown -R verdaccio:verdaccio /verdaccio
```

I think the problem might be that we specify the volume after that command and after assuming the verdaccio user.
```
VOLUME ["/verdaccio"]
```

**Comment by markpeterfejes** [comment]
I've run into the same problem as above. This is the`ls -la` output from the `/verdaccio/storage` folder:

```
/verdaccio/storage $ ls -la
total 12
drwxr-xr-x    2 verdacci verdacci      4096 Jul  4 08:56 .
drwxr-xr-x    4 verdacci verdacci      4096 Jul  4 08:56 ..
-rw-r--r--    1 verdacci nogroup         87 Jul  4 08:56 .sinopia-db.json
```

**Comment by markpeterfejes** [comment]
That's not the problem unfortunatelly. It won't help. Until the problem will be found, the following unsecure workaround can be used, add this to your deployment right before the verdaccio container spec:

```json
{
    "initContainers": [
        {
            "name": "verdaccio-init",
            "image": "busybox",
            "imagePullPolicy": "IfNotPresent",
            "command": [
                "sh",
                "-c",
                "chmod 777 -R  /verdaccio/storage"
            ],
            "volumeMounts": [
                {
                    "name": "verdaccio-storage",
                    "mountPath": "/verdaccio/storage"
                }
            ]
        }
    ]
}```

**Comment by juanpicado** [comment]
Can that can be tested in a local environment?

**Comment by markpeterfejes** [comment]
I think you could use minikube to test it locally. Kubernetes mounts the volume just before the dockerfile's CMD command as root. Thats why we cannot write to the storage folder.

**Comment by juanpicado** [comment]
Cool, thanks for that 👍

**Comment by markpeterfejes** [comment]
This is not a kubernetes specific problem unfortunately. If you run the following command the image is going to fail with EPERM, because docker is mounting the volumes as the host user or root.
`docker run -it -v ~/test:/verdaccio/storage -u verdaccio verdaccio/verdaccio:latest`

The issue is tracked [here](https://github.com/moby/moby/issues/2259).

**Comment by markpeterfejes** [comment]
@adrianchifor So to recap this, the fix for this problem is to set this on the PodSpec:

```json
"securityContext": {
    "fsGroup": 101
}                        
```

Where 101 is the verdaccio group id. Kubernetes is going to give permission for that group to write on the attached volume

**Comment by juanpicado** [comment]
Shipped under `v2.2.5`. Docker image will be ready soon.

**Comment by adrianchifor** [comment]
@markpeterfejes @juanpicado cheers guys!

**Comment by juanpicado** [comment]
I wonder if this fix on `npm 5.1.0` is related with this 
Releases notes: 

> https://github.com/npm/npm/commit/58be2ec596dfb0353ad2570e6750e408339f1478  Make sure uid and gid are getting correctly set even when they're 0. This should fix some Docker-related issues with bad permissions/broken ownership. (@rgrove)
(@zkat)

https://github.com/npm/npm/commit/24ec9f2dd4dcd4f25622dff3a502d4e86a025c0d
